app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java

```diff
@@ -1,27 +1,67 @@
 package org.sagebionetworks.bridge.services;
 
+import org.apache.commons.lang3.StringUtils;
+import org.sagebionetworks.bridge.crypto.AesGcmEncryptor;
 import org.sagebionetworks.bridge.models.User;
 import org.sagebionetworks.bridge.models.UserProfile;
 
 import com.stormpath.sdk.account.Account;
+import com.stormpath.sdk.directory.CustomData;
 
 public class UserProfileServiceImpl implements UserProfileService {
 
     private AuthenticationService authService;
     
+    private AesGcmEncryptor healthCodeEncryptor;
+
     public void setAuthenticationService(AuthenticationService authService) {
         this.authService = authService;
     }
+    
+    public void setHealthCodeEncryptor(AesGcmEncryptor encryptor) {
+        this.healthCodeEncryptor = encryptor;
+    }
 
     @Override
+    public UserProfile getProfile(String email) {
+        Account account = authService.getAccount(email);
+
+        UserProfile profile = new UserProfile();
+        profile.setFirstName(removeEmpty(account.getGivenName()));
+        profile.setLastName(removeEmpty(account.getSurname()));
+        profile.setUsername(removeEmpty(account.getUsername()));
+        profile.setEmail(removeEmpty(account.getEmail()));
+        
+        String encryptedPhone = (String)account.getCustomData().get("phone");
+        if (encryptedPhone != null) {
+            profile.setPhone(healthCodeEncryptor.decrypt(encryptedPhone));
+        }
```

Comment:
If phone number isn't specified, the field will return null. If name isn't specified, the field will return empty string. Should we change phone number to empty string or removeEmpty to null to be consistent?